rfcs/react-components/styles-handbook.md

```diff
@@ -0,0 +1,921 @@
+import { tokens } from '@fluentui/react-theme';
+
+const useClasses = makeStyles({
+  slotA: {
+    backgroundColor: tokens.colorNeutralBackground1,
+    // ❌ Don't do
+    // You can apply classes directly to that "div"
+    '> div': {
+      color: tokens.colorNeutralForeground1,
+    },
+  },
+
+  // ✅ Do
+  slotA: {
+    backgroundColor: tokens.colorNeutralBackground1,
+  },
+  slotB: {
+    color: tokens.colorNeutralForeground1,
+  },
+});
+
+function App(props) {
+  const classes = useClasses();
+
+  return (
+    <div className={classes.root}>
+      {/* 💡 Classes can be passed directly to this element */}
+      <div className={classes.slot} />
+    </div>
+  );
+}
+```
+
+### Avoid complicated selectors
+
+Keep selectors simple to produce reusable CSS rules:
+
+- CSS rules that are unique cannot be reused in other areas
+  ```css
+  /*    ⬇️ cannot be reused in other components  */
+  .hash .some-unique-class {
+    display: flex;
+  }
+  ```
+- Complicated selectors produce bigger bundle size with AOT
+
+  ```js
+  makeStyles({
+    rootA: {
+      display: 'flex',
+    },
+    rootB: {
+      '> .some-classname': {
+        '> .other-classname': {
+          display: 'flex',
+          alignItems: 'center',
+        },
+      },
+    },
+  });
+  ```
+
+  ⬇️⬇️⬇️
+
+  ```css
+  /* ✅ no selectors */
+  .f22iagw {
+    display: flex;
+  }
+
+  /* ⚠️ with complex selectors */
+  .f1312jvm > .some-classname > .other-classname {
+    display: flex;
+  }
+  .f1c58nry > .some-classname > .other-classname {
+    align-items: center;
+  }
+  ```
+
+- Complicated selectors are hard to override as overrides should match component's styles
+
+  ```js
+  // On component side
+  makeResetStyles({
+    '> .some-classname': {
+      '> .other-classname': {
+        ':hover': {
+          display: 'flex',
+          alignItems: 'center',
+        },
+      },
+    },
+  });
+  // On consumer side 💥
+  makeStyles({
+    foo: {
+      '> .some-classname > .other-classname:hover': {
+        display: 'flex',
+        alignItems: 'center',
+      },
+    },
+  });
+  ```
+
+```js
+import { makeStyles, makeResetStyles, mergeClasses, shorthands } from '@griffel/react';
+import { tokens } from '@fluentui/react-theme';
+
+// ❌ Don't do
+// Avoid complex selectors i.e. simplify them
+const useBaseClassName = makeResetStyles({
+  '> .foo-classname': {
+    '> .bar-classname': {
+      '> .baz-classname': {
+        display: 'flex',
+        alignItems: 'center',
+      },
+    },
+  },
+});
+
+// ✅ Do
+// Apply classes directly to an element
+const useBaseBazClasses = makeResetStyles({
+  display: 'flex',
+  alignItems: 'center',
+});
+```
+
+### Avoid input pseudo classes
+
+Instead of usage [input pseudo classes][mdn-input-pseudo-classes] in styles, prefer to use JS state.
+
+- Produces less classes on an element
+- Selectors for overrides are simpler
+- These pseudo classes are supported by input elements
```

Comment:
I get the point in L866 and L867. Not sure what L868 has anything to do with the point of preferring js state over pseudo class?

Reply:
I missed "only" there, it should be "These pseudo classes are only supported by input elements"